neurolib/models/model.py

```diff
@@ -2,34 +2,321 @@
+            if svn in self.output_vars:
+                self.setOutput(svn, sv, append=append_outputs)
+            self.setStateVariables(svn, sv)
+
+    def integrate_chunkwise(self, chunksize, bold=False, append_outputs=False):
+        """Repeatedly calls the chunkwise integration for the whole duration of the simulation.
+        If `bold==True`, the BOLD model is simulated after each chunk.     
+        
+        :param chunksize: size of each chunk to simulate in units of dt
+        :type chunksize: int
+        :param bold: simulate BOLD model after each chunk, defaults to False
+        :type bold: bool, optional
+        :param append_outputs: append the chunkwise outputs to the outputs attribute, defaults to False
+        :type append_outputs: bool, optional
+        """
+        totalDuration = self.params["duration"]
+
+        dt = self.params["dt"]
+        # create a shallow copy of the parameters
+        lastT = 0
+        while lastT < totalDuration:
+            # Determine the size of the next chunk
+            currentChunkSize = min(chunksize, (totalDuration - lastT) / dt)
+            currentChunkSize += self.max_delay + 1
+
+            self.autochunk(duration=currentChunkSize * dt, append_outputs=append_outputs)
+
+            if bold and self.bold_initialized:
+                self.simulate_bold()
+
+            # we save the last simulated time step
+            lastT += self.state["t"][-1]
+
+    def autochunk(self, inputs=None, duration=None, append_outputs=False):
+        """Executes a single chunk of integration, either for a given duration
+        or a single timestep `dt`. Gathers all inputs to the model and resets
+        the initial conditions as a preparation for the next chunk. 
+        
+        :param inputs: list of input values, ordered according to self.input_vars, defaults to None
+        :type inputs: list[np.ndarray|], optional
+        :param duration: length of a chunk to simulate in ms, defaults to a single dt, defaults to None
+        :type duration: float, optional
+        :param append_outputs: append the chunkwise outputs to the outputs attribute, defaults to False
+        :type append_outputs: bool, optional
+        """
+        startindt = self.max_delay + 1
+        if duration is not None:
+            chunkDuration = duration
+        else:
+            chunkDuration = startindt * self.params["dt"] + self.params["dt"]
+        self.params["duration"] = chunkDuration
+        # set inputs
+        if inputs is not None:
+            for i, iv in enumerate(self.input_vars):
+                self.params[iv] = inputs[i]
+
+        # run integration
+        self.integrate(append_outputs=append_outputs)
+
+        # reset initial conditions to last state
+        for iv, sv in zip(self.init_vars, self.state_vars):
+            # if state variables are one-dimensional (in space only)
+            if len(self.state[sv].shape) == 1:
```

Comment:
```suggestion
            if self.state[sv].ndim == 1:
```

Reply:
This tip improved my life by `0.01%`, thank you. Always been doing it wrong.